File: 0135-candy/0135-candy.py
class Solution:
    def candy(self, ratings: List[int]) -> int:
        size = len(ratings)
        left_arr = [0 for i in range(size)]
        left_arr[0] = 1
        total_answer = 0
        right,curr = 1,0
        for i in range(1,size):
            if ratings[i] > ratings[i - 1]:
                left_arr[i] = left_arr[i - 1] + 1
            else:
                left_arr[i] = 1

        for i in range(size - 2,-1,-1):
            if ratings[i] > ratings[i + 1]:
                curr = right + 1
            else:
                curr = 1
            total_answer += max(curr,left_arr[i])
            right = curr
        total_answer += max(1,left_arr[-1])
        return total_answer

File: 0135-candy/0135-candy.java
class Solution {
    public int candy(int[] ratings) {
        int size = ratings.length;
        int [] left = new int[size];
        int [] right = new int[size];
        left[0] = 1;
        right[size - 1] = 1;

        for(int i = 1; i < size; i++){
            if(ratings[i] > ratings[i - 1]){
                left[i] = left[i - 1] + 1;
            }
            else{
                left[i] = 1;
            }
        }

        for(int j = size - 2; j >= 0; j--){
            if(ratings[j] > ratings[j + 1]){
                right[j] = right[j + 1] + 1;
            }
            else{
                right[j] = 1;
            }
        }
        int answer = 0;
        for(int index = 0;index < size; index++){
            answer += Math.max(left[index], right[index]);
        }
        return answer;
           
    }
}